cloudflare/resource_cloudflare_worker_secret.go

```diff
@@ -0,0 +1,119 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareWorkerSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWorkerSecretCreate,
+		Read:   resourceCloudflareWorkerSecretRead,
+		Delete: resourceCloudflareWorkerSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareWorkersSecretImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"script_name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The name of the Worker script to associate the secret with.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The name of the Worker secret.",
+			},
+			"secret_text": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "The text of the Worker secret, this cannot be read back after creation and is stored encrypted .",
+			},
+		},
+	}
+}
+
+func resourceCloudflareWorkerSecretRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	scriptName := d.Get("script_name").(string)
+	name := d.Get("name").(string)
+
+	secrets, err := client.ListWorkersSecrets(context.Background(), scriptName)
+	if err != nil {
+		return errors.Wrap(err, "error reading worker secrets")
+	}
+
+	for _, secret := range secrets.Result {
+		if secret.Name == name {
+			return nil
+		}
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceCloudflareWorkerSecretCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	scriptName := d.Get("script_name").(string)
+	name := d.Get("name").(string)
+	secretText := d.Get("secret_text").(string)
+
+	request := cloudflare.WorkersPutSecretRequest{
+		Name: name,
+		Text: secretText,
+		Type: cloudflare.WorkerSecretTextBindingType,
+	}
+
+	_, err := client.SetWorkersSecret(context.Background(), scriptName, &request)
+	if err != nil {
+		return errors.Wrap(err, "error creating worker secret")
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", scriptName, name))
```

Comment:
lets use [`stringChecksum`](https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/utils.go#L61-L67) here to output a MD5 and match the other resource IDs

 ```suggestion
	d.SetId(stringChecksum(fmt.Sprintf("%s/%s", scriptName, name)))
```